presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -238,10 +243,17 @@ protected PlanNodeStatsEstimate visitComparisonExpression(ComparisonExpression n
                 return process(new ComparisonExpression(type.flip(), right, left));
             }
 
-            if (left instanceof SymbolReference && right instanceof Literal) {
-                Symbol symbol = Symbol.from(left);
-                OptionalDouble literal = doubleValueFromLiteral(types.get(symbol), (Literal) right);
-                return comparisonSymbolToLiteralStats(input, symbol, literal, type);
+            if (left instanceof Literal && !(right instanceof Literal)) {
+                // normalize so that literal is on the right
+                return process(new ComparisonExpression(type.flip(), right, left));
+            }
+
+            if (right instanceof Literal) {
+                // TODO support Cast(Literal) same way as Literal (nested Casts too)
```

Comment:
maybe we could leverage `com.facebook.presto.sql.planner.ExpressionInterpreter#evaluateConstantExpression`? We would need to make it return `Optional`